sonarts-core/src/runner/diagnostics.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * SonarTS
+ * Copyright (C) 2017-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+import * as ts from "typescript";
+import { lineAndCharacterByPos } from "../utils/navigation";
+import { toSonarLine } from "./sonarUtils";
+
+export default function getDiagnostics(sourceFile: ts.SourceFile, program: ts.Program): any[] {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Type declaration of 'any' loses type-safety. Consider replacing it with a more precise type. ([typescript:S4204](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS4204))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=sonarts&pullRequest=637&issues=AWNixfAfdQHZQLlXBUrl&open=AWNixfAfdQHZQLlXBUrl)

[ISSUE_KEY]: AWNixfAfdQHZQLlXBUrl